### PR TITLE
Make @QuarkusTest play nicely with @ParameterizedTest

### LINKED ISSUE #7877: Unable to use @ParameterizedTest with test annotated by @QuarkusTest
I was building camel-quarkus with the latest Quarkus SNAPSHOT and noticed a change in behaviour. Tests annotated with `@QuarkusTest` that use `@ParameterizedTest` have started to fail with:

```
org.junit.jupiter.api.extension.ParameterResolutionException: Discovered multiple competing ParameterResolvers for parameter [java.lang.String arg0] in method [public void org.apache.camel.quarkus.component.dataformats.json.JsonComponentsTest.testUnmarshal(java.lang.String)]: io.quarkus.test.junit.QuarkusTestExtension@36e9794c, org.junit.jupiter.params.ParameterizedTestParameterResolver@78a0d664
```

Here's an example failing test:

https://github.com/apache/camel-quarkus/blob/master/integration-tests/dataformats-json/src/test/java/org/apache/camel/quarkus/component/dataformats/json/JsonComponentsTest.java

### PR DESCRIPTION
**Pull request by geoand**
Fixes: #7877

### CODE REVIEW

File: test-framework/junit5/src/main/java/io/quarkus/test/junit/QuarkusTestExtension.java [hunk 1]
```diff
@@ -365,12 +365,12 @@ public void afterAll(ExtensionContext context) throws Exception {
     }
 
     /**
-     * By returning true, we allow a QuarkusTest to use constructor injection
+     * Return true if we are need a parameter for constructor injection
```
Comment (gsmet) [inline test-framework/junit5/src/main/java/io/quarkus/test/junit/QuarkusTestExtension.java]:
```suggestion
     * Return true if we need a parameter for constructor injection
```

**Review by gsmet — CHANGES_REQUESTED** [review]
Spotted a small typo. Otherwise looks good.

**Comment by gsmet** [comment]
@geoand I added the backport label? I think we would want to backport this, no? I have a vague recollection this went into 1.3.

Please remove the label if I'm wrong!

**Review by gastaldi — APPROVED** [review]
LGTM

**Comment by geoand** [comment]
@gsmet it was not backported to 1.3: https://github.com/quarkusio/quarkus/pull/7714

File: test-framework/junit5/src/main/java/io/quarkus/test/junit/QuarkusTestExtension.java (hunk 1 above)
Comment (geoand) [reply test-framework/junit5/src/main/java/io/quarkus/test/junit/QuarkusTestExtension.java]: Thanks, fixed!

**Comment by gsmet** [comment]
Ah yes, now I remember, you told me it wasn't critical. Thanks for correcting me!

**Review by gsmet — APPROVED** [review]

**Comment by geoand** [comment]
Thanks for being thorough :wink:

**Comment by geoand** [comment]
I actually added the backport label to both this and #7714

**Pull request merged**